quantum/rgblight.c

```diff
@@ -628,7 +628,7 @@ void rgblight_sethsv_slave(uint8_t hue, uint8_t sat, uint8_t val) { rgblight_set
 
 #ifdef RGBLIGHT_LAYERS
 void rgblight_set_layer_state(uint8_t layer, bool enabled) {
-    rgblight_layer_mask_t mask = 1 << layer;
+    rgblight_layer_mask_t mask = (rgblight_layer_mask_t)1UL << layer;
```

Comment:
If we're already casting to `rgblight_layer_mask_t`, surely the UL suffix is not needed? Otherwise it is always 32 bits wide which may result in suboptimal code.

Reply:
I just checked; it works without the cast, and saves 54 bytes.

Reply:
What about the other way around?

Reply:
It seems to work with the cast too, and that saves an additional 48 bytes.